pkg/webhook/v1alpha3/experiment/validator/validator.go

```diff
@@ -208,5 +211,41 @@ func (g *DefaultValidator) validateMetricsCollector(inst *experimentsv1alpha3.Ex
 	if mcjob.GetNamespace() != namespace || mcjob.GetName() != trialName {
 		return fmt.Errorf("Invalid metricsCollector template.")
 	}
+
+	// Above part of this method will be dropped
+	mcSpec := inst.Spec.MetricsCollectorSpec
+	mcKind := mcSpec.Collector.Kind
+	// TODO(hougangliu):
```

Comment:
Should we remove the TODO here?

Reply:
yes, we need it since we'd better enhance here in future